### PR TITLE
Use TOS lingo and table format descriptions

### PR DESCRIPTION
**Pull request by rocky**
The way I am currently thinking about describing instructions. 

Thoughts?

### CODE REVIEW

File: elisp-bytecode.texi [hunk 1]
```diff
@@ -1041,55 +1041,220 @@ bytecode interpreter.
...
+@end table
 
 @node eqlsign
 @unnumberedsubsubsec @code{eqlsign} (85)
 @kindex eqlsign
-Call @code{=} with two arguments, comparing the two values at the top of the stack for numerical or strict equality.
+
+@table @strong
+@item Implements:
+@math{TOS = TOS == 1}.
```
Comment (pipcet) [inline elisp-bytecode.texi]: I think using just 1 here is confusing and ambiguous given the descriptions of 1+ and 1-. Also, `==` is the C operator that returns 0 or 1, not `nil` or `t`. So overall I'd prefer something like
`TOS <- (= TOS S[1])`, keeping the Lisp syntax.

File: elisp-bytecode.texi [hunk 2]
```diff
@@ -1041,55 +1041,220 @@ bytecode interpreter.
...
 @node plus
 @unnumberedsubsubsec @code{plus} (92)
 @kindex plus
-Call @code{+} with two arguments, adding the two values at the top of the stack.
 
-@strong{Example}
-When dynamic binding is in effect, @code{(defun plus-eg(n) (+ n n ))} generates:
+@table @strong
+@item Implements:
+@math{TOS = TOS + TOS-1}.
```
Comment (pipcet) [inline elisp-bytecode.texi]: This is different syntax from what's used above.

File: elisp-bytecode.texi [hunk 3]
```diff
@@ -1117,30 +1291,97 @@ PC  Byte  Instruction
...
+binary @code{max}.
+@item Stack effect:
+@math{-2+1}.
+@item Example:
+When lexical binding is in effect, @code{(defun max-eg(a b) (max a b))} generates:
+@verbatim
+PC  Byte  Instruction
+ 0    1   stack-ref[1]
+ 1    1   stack-ref[1]
+ 2   86   diff
```
Comment (pipcet) [inline elisp-bytecode.texi]: This should be `max`, not `diff`.

File: elisp-bytecode.texi [hunk 4]
```diff
@@ -1117,30 +1291,97 @@ PC  Byte  Instruction
...
+
+@table @strong
+@item Implements:
+@math{TOS = TOS \% TOS-1)}.
+@item Generated via:
+binary @verb{|%|}.
+@item Stack effect:
+@math{-2+1}.
+@item Example:
+When lexical binding is in effect, @code{(defun min-quo(a b) (quo a b))} generates:
```
Comment (pipcet) [inline elisp-bytecode.texi]: I have no `quo` function, just `/`...

File: elisp-bytecode.texi [hunk 5]
```diff
@@ -1117,30 +1291,97 @@ PC  Byte  Instruction
...
+@end table
 
 @node quo
 @unnumberedsubsubsec @code{quo} (165)
 @kindex quo
-Call @code{/} with two arguments, dividing the two values at the top of the stack.
+
+@table @strong
+@item Implements:
+@math{TOS = TOS \% TOS-1)}.
```
Comment (pipcet) [inline elisp-bytecode.texi]: `/` rather than `%`, I think.

File: elisp-bytecode.texi [hunk 6]
```diff
@@ -1117,30 +1291,97 @@ PC  Byte  Instruction
...
 
 @node rem
 @unnumberedsubsubsec @code{rem} (166)
 @kindex rem
-Call @code{%} with two arguments, calculating the modulus of the two values at the top of the stack.
 
-Added in Emacs 19.34.
+@table @strong
+@item Implements:
+@math{TOS = TOS / TOS-1)}.
```
Comment (pipcet) [inline elisp-bytecode.texi]: And `%` here

File: elisp-bytecode.texi [hunk 7]
```diff
@@ -1117,30 +1291,97 @@ PC  Byte  Instruction
...
+@item Generated via:
+binary @verb{|/|}
+@item Stack effect:
+@math{-2+1}
+@item Added in:
+Emacs 19.34.
+@item Example:
+When lexical binding is in effect, @code{(defun rem-eg(a b) (rem a b))} generates:
```
Comment (pipcet) [inline elisp-bytecode.texi]: I have no `rem`, just `%`.

File: elisp-bytecode.texi (hunk 2 above)
Comment (pipcet) [reply elisp-bytecode.texi]: Also, this reads (to me) like it doesn't pop a value, when in fact it does.

Comment (rocky) [reply elisp-bytecode.texi]: Just revised and see if this is better. 

All of this is a bit tedious and error prone. I think I got all of the things mentioned, but I might have missed stuff. Feel free just to make corrections in place. 

(And there is a lot more that we haven't gotten to). 

I think the return instruction, should be lumped in with control-flow instructions, e.g. jump and jumpif... which we haven't gotten around to yet.

**Comment by rocky** [comment]
The main purpose here was to suggest a different way to describe instructions. Not fill in all of the instructions per se. (That I think will take a bit of time to complete and my thought is to just go section by section. The control flow instructions which we really don't have anything on except possibly "return", will probably be next.)

So unless there are more corrections or thoughts about the overall method of describing instructions in the existing two sections, let's merge this and do the rest section by section.

**Comment by pipcet** [comment]
Sure, let's merge this!

**Pull request merged**